pkg/apis/eks.cattle.io/v1/types.go

```diff
@@ -33,63 +33,111 @@ type EKSClusterConfig struct {
-	EBSCSIDriver           *bool             `json:"ebsCSIDriver"`
-	PublicAccessSources    []string          `json:"publicAccessSources"`
-	LoggingTypes           []string          `json:"loggingTypes"`
-	Subnets                []string          `json:"subnets" norman:"noupdate"`
-	SecurityGroups         []string          `json:"securityGroups" norman:"noupdate"`
-	ServiceRole            *string           `json:"serviceRole" norman:"noupdate,pointer"`
-	NodeGroups             []NodeGroup       `json:"nodeGroups"`
+	// AmazonCredentialSecret is the name of the secret containing the Amazon credentials.
+	AmazonCredentialSecret string `json:"amazonCredentialSecret"`
+	// DisplayName is the name of the cluster to be displayed in the UI.
+	DisplayName string `json:"displayName" norman:"noupdate"`
+	// Region is the AWS region to create the cluster in.
+	Region string `json:"region" norman:"noupdate"`
+	// Imported is true if the cluster was imported.
+	Imported bool `json:"imported" norman:"noupdate"`
+	// KubernetesVersion is the version of Kubernetes to use.
+	KubernetesVersion *string `json:"kubernetesVersion" norman:"pointer"`
+	// Tags is a map of tags to apply to the cluster.
+	Tags map[string]string `json:"tags"`
+	// SecretsEncryption is true if secrets should be encrypted.
+	SecretsEncryption *bool `json:"secretsEncryption" norman:"noupdate"`
+	// KmsKey is the KMS key to use for encryption.
+	KmsKey *string `json:"kmsKey" norman:"noupdate,pointer"`
+	// PublicAccess is true if the cluster should be publicly accessible.
+	PublicAccess *bool `json:"publicAccess"`
+	// PrivateAccess is true if the cluster should be privately accessible.
+	PrivateAccess *bool `json:"privateAccess"`
+	// EbsCSIDriver is true if the EBS CSI driver should be installed.
+	EBSCSIDriver *bool `json:"ebsCSIDriver"`
+	// PublicAccessSources is a list of CIDRs that can access the cluster.
+	PublicAccessSources []string `json:"publicAccessSources"`
+	// LoggingTypes is a list of logging types to enable.
+	LoggingTypes []string `json:"loggingTypes"`
+	// Subnets is a list of subnets to use for the cluster.
+	Subnets []string `json:"subnets" norman:"noupdate"`
+	// SecurityGroups is a list of security groups to use for the cluster.
+	SecurityGroups []string `json:"securityGroups" norman:"noupdate"`
+	// ServiceRole is the IAM role to use for the cluster.
+	ServiceRole *string `json:"serviceRole" norman:"noupdate,pointer"`
+	// NodeGroups is a list of node groups to create.
+	NodeGroups []NodeGroup `json:"nodeGroups"`
 }
 
 type EKSClusterConfigStatus struct {
-	Phase                         string            `json:"phase"`
-	VirtualNetwork                string            `json:"virtualNetwork"`
-	Subnets                       []string          `json:"subnets"`
-	SecurityGroups                []string          `json:"securityGroups"`
-	ManagedLaunchTemplateID       string            `json:"managedLaunchTemplateID"`
+	// Phase is the current lifecycle phase of the cluster.
```

Comment:
It would also be good to add doc strings to the struct definitions.

Reply:
fixed